File: cases/taint/java/call-context-negative/CallContextNegative.java
package dataflowbench.taint;

final class CallContextNegative {
    static int dfb_source() { // DFB-SOURCE: call-context-input
        return 1;
    }

    static int relay(int value) { // DFB-WITNESS: call-context-relay
        return value;
    }

    static void dfb_sink(int value) { } // DFB-SINK: call-context-sink

    static void run() {
        int tainted = relay(dfb_source());
        int clean = relay(0);
        dfb_sink(clean);
    }
}

File: cases/taint/go/call-context-negative/call_context_negative.go
package dataflowbench

func dfb_source() int { // DFB-SOURCE: call-context-input
	return 1
}

func relay(value int) int { // DFB-WITNESS: call-context-relay
	return value
}

func dfb_sink(value int) {} // DFB-SINK: call-context-sink

func run() {
	_ = relay(dfb_source())
	clean := relay(0)
	dfb_sink(clean)
}
